setup.py

```diff
@@ -244,8 +244,20 @@ def run(self):
 
 # Qt bindings requirements
 qt_requirements = {
-    'pyqt5': ['pyqt5>=5.15,<5.16', 'pyqtwebengine>=5.15,<5.16', 'pyqt5-sip<12.16; python_version=="3.8"'],
-    'pyqt6': ['pyqt6>=6.5,<7', 'pyqt6-webengine>=6.5,<7'],
+    'pyqt5': [
+        'pyqt5>=5.15,<5.16',
+        'pyqt5-sip<12.16; python_version=="3.8"',
+        'pyqtwebengine>=5.15,<5.16',
+        'qtconsole>=5.6.1,<5.7.0',
+    ],
+    'pyqt6': [
+        'pyqt6>=6.5,<7',
+        'pyqt6-webengine>=6.5,<7',
+        'qtconsole>=5.6.1,<5.7.0',
+    ],
+    'conda-forge': [
+        'qtconsole>=5.6.1,<5.7.0',
```

Comment:
Shouldn't this be `qtconsole-base` instead of `qtconsole`, as you have it before?

Otherwise, qtconsole will bring `pyqt5`, won't it?

Reply:
So this is a bit nuanced.
When `qtconsole-base` is installed by conda, `conda list` shows `qtconsole-base 5.6.1`, but `pip list` shows `qtconsole 5.6.1`. If `setup.py` specifies `qtconsole-base`, `pip --check` fails with `spyder 6.1.0a1.dev0 requires qtconsole-base, which is not installed.`. See [this pipline failure](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1104144&view=logs&j=535b614b-9e28-51ef-eb67-d0aa83d01d61&t=f3b82646-01b4-574c-79b5-1561965b3e04&l=8355).

This will only introduce `pyqt5` _if_ Spyder is built with `SPYDER_QT_BINDING=conda-forge` _and_ the resulting package is installed using pip. I don't know how to install `qtconsole-base` and also have pip see it as such.

Reply:
Understood, thanks for the clarification @mrclary! Let's leave things like this then.